java/com/fortech/internship/service/impl/CityService.java

```diff
@@ -0,0 +1,87 @@
+package com.fortech.internship.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.fortech.internship.dao.AddressDAO;
+import com.fortech.internship.dao.CityDAO;
+import com.fortech.internship.dto.AddressDTO;
+import com.fortech.internship.dto.CityDTO;
+
+import com.fortech.internship.model.Address;
+import com.fortech.internship.model.City;
+import com.fortech.internship.model.User;
+
+
+@Service
+
+public class CityService {
+	
+	@Autowired
+	private AddressDAO addressDAO;
+	
+	@Autowired
+	private CityDAO cityDAO;
+	
+	@Autowired
+	private SessionFactory sessionFactory;
+
+	@Transactional
+	public void createCity(CityDTO cityDTO) {
+		//cityDAO.createCity(cityDTO);
+		Session session = sessionFactory.getCurrentSession();
+		City city = new City();
+		city.setName(cityDTO.getName());
+		city.setAddress(sessionFactory.getCurrentSession().get(Address.class, cityDTO.getAddressDTO().getId()));
+		session.save(city);	
+	}
+
+	@Transactional
```

Comment:
on "get" methods you can use @Transactional(ReadOnly=true) because spring does some performance improvements in backend

Reply:
solved, please check